plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/MachineProviderImpl.java

```diff
@@ -515,6 +518,17 @@ private String createContainer(String workspaceId,
                                            .collect(toMap(Function.identity(),
                                                           value -> new PortBinding[0])))
                   .withVolumesFrom(toArrayIfNotNull(service.getVolumesFrom()));
+        
+        if (volumesFrom != null) {
```

Comment:
Yes, looks like you've added test correctly despite complexity of test framework. Do you have any suggestion how to make it less confusing?